File: s3-inf-ppo/laboratoria/lab10/php/src/Entries/DrankBeer.php
<?php

namespace PPO\Notebook\Entries;

use PPO\Notebook\Interfaces\NotebookEntry;
use PPO\Notebook\Traits\Slugger;

class DrankBeer implements NotebookEntry {

	use Slugger;

	private $breweryName;
	private $name;

	public function __construct(string $name, string $breweryName = "") {
		$this->name = $name;
		$this->breweryName = $breweryName;
	}

	protected function getSlugBase(): string {
		return $this->breweryName . " " . $this->name;
	}

}

File: s3-inf-ppo/laboratoria/lab10/java/src/Notebook/Entries/DrankBeer.java
package Notebook.Entries;

import Notebook.Interfaces.NotebookEntry;
import Notebook.Traits.Slugger;

public class DrankBeer implements NotebookEntry, Slugger {

    private String name;
    private String brewery;

    public DrankBeer(String name, String brewery) {
        this.name = name;
        this.brewery = brewery;
    }

    @Override
    public String getSlug() {
        return getSlug(brewery + " " + name);
    }

}
